apollo-integration/src/test/java/com/apollographql/apollo/performance/NumberParsingTest.kt

```diff
@@ -0,0 +1,103 @@
+package com.apollographql.apollo.performance
+
+import com.apollographql.apollo.ApolloClient
+import com.apollographql.apollo.Utils.immediateExecutor
+import com.apollographql.apollo.Utils.immediateExecutorService
+import com.apollographql.apollo.api.internal.SimpleOperationResponseParser
+import com.apollographql.apollo.fetcher.ApolloResponseFetchers
+import com.apollographql.apollo.integration.performance.GetFloatsQuery
+import com.apollographql.apollo.integration.performance.GetIntsQuery
+import kotlinx.coroutines.runBlocking
+import kotlinx.serialization.json.JsonArray
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonPrimitive
+import okhttp3.Dispatcher
+import okhttp3.OkHttpClient
+import okio.ByteString
+import okio.ByteString.Companion.encodeUtf8
+import kotlin.random.Random
+import kotlin.test.Test
+import kotlin.time.ExperimentalTime
+import kotlin.time.measureTime
+
+data class Data(
+    val randomInts: List<Int>
+)
+
+/**
+ * These tests are disabled by default. Set the `runPerformanceTests` property to run them:
+ * ./gradlew :apollo-integration:testDebug --tests '*parseFloats*' -DrunPerformanceTests=true
+ */
+@OptIn(ExperimentalTime::class)
+class NumberParsingTest {
+  /**
+   * A test to benchmark the parsing of integers in Json.
+   */
+  @Test
+  fun parseInts() {
+
+    val random = Random.Default
+
+    val data = JsonObject(
+        mapOf(
+            "data" to JsonObject(
+                mapOf(
+                    "randomInts" to JsonArray(
+                        0.until(10000).map {
+                          JsonPrimitive(random.nextInt())
+                        }
+                    )
+                )
+            )
+        )
+    )
+
+    val json = data.toString().encodeUtf8()
```

Comment:
Is there any reason to not use our `JsonWriter` to compose thus test json? I mean at least we can avoid extra dependency to `kotlinx-serialization` that is used as far as I understood only for this test?

Reply:
It was mostly me being lazy to count the number of `beginObject` and `endObjects` using the `JsonWriter` API. I just did it.